libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -298,9 +298,38 @@ void CompilerUtils::zeroInitialiseMemoryArray(ArrayType const& _type)
 	m_context << Instruction::SWAP1 << Instruction::POP;
 }
 
-void CompilerUtils::memoryCopy()
+void CompilerUtils::memoryCopy(bool _useIdentityPrecompile)
 {
 	// Stack here: size target source
+
+	if (!_useIdentityPrecompile)
+	{
+		m_context.appendInlineAssembly(R"(
+			{
+			// expects three locals: src, dst, len
+
+			// copy 32 bytes at once
+			start32:
+			jumpi(end32, lt(len, 32))
+			mstore(dst, mload(src))
+			dst := add(dst, 32)
+			src := add(src, 32)
+			len := sub(len, 32)
+			jump(start32)
+			end32:
+
+			// copy the last byte
+			let mask := sub(exp(256, sub(32, len)), 1)
+			let srcpart := and(mload(src), not(mask))
+			let dstpart := and(mload(dst), mask)
+			mstore(dst, or(srcpart, dstpart))
+			}
+		)",
+			{ "src", "dst", "len" }
```

Comment:
If that was the original comment, then the assignment should be `{"len", "dst', "src"}`